src/lib/utilities/Dialog/stores.ts

```diff
@@ -2,30 +2,38 @@
 
 import { writable } from 'svelte/store';
 
-export interface DialogAlert {
-	icon?: string;
-	title: string;
-	body: string;
-	image?: any;
-	component?: any;
-	html?: any;
-}
-export interface DialogConfirm extends DialogAlert {
-	type: string;
-	result: any;
+export interface DialogComponent {
+	/** Import and provide your component reference. */
+	ref: any;
+	/** Provide a key/value pairs of component props. */
+	props?: Record<string, string>;
+	/** Provide an HTML template literal for the default slot. */
+	slot?: string;
 }
-export interface DialogPrompt extends DialogAlert {
-	type: string;
-	value: any;
-	result: any;
+
+export interface Dialog {
+	/** Designate what type of component will display. */
+	type: 'alert' | 'confirm' | 'prompt' | 'component';
+	/** Provide the dialog header content. Accepts HTML. */
+	title?: string;
+	/** Provide the dialog body content. Accepts HTML. */
+	body?: string;
+	/** Provide a URl to display an image within the dialog. */
+	image?: any;
+	/** Prompt Only - provide a prompt value. */
+	value?: any;
+	/** Provide {component, props} to create a generate a custom component. */
+	component?: DialogComponent;
+	/** Provide a function. Returns the repsonse value. */
+	response?: any;
 }
 
 function dialogService(): any {
 	const { subscribe, set, update } = writable([]);
 	return {
 		subscribe,
 		// Trigger - append to end of queue
-		trigger: (dialog: DialogAlert | DialogAlert | DialogAlert) =>
+		trigger: (dialog: Dialog) =>
 			update((dStore: any) => {
```

Comment:
The preview doesn't highlight which line you're referring to, but I'm assume it's the last. The `dStore: any`, correct?

None of the following worked without error:

- `any[]`
- `DialogSettings[]`
- removing the type

Thoughts?

Reply:
This function has a couple issues that would be solved with some more rigorous ESLinting- you're still popping in `any` typings. There is virtually no situation where we should be letting `any`s slip into prod. When you really have to, prefer `unknown`, but it's turning off typescript for lots of code. Additionally, if you use JSDoc syntax, intellisense will provide the comments for the `close`, `trigger`, and `clear` functions which are very useful.

I don't wanna get my tree messy, but here's how it might look: 

```ts
// Dialog Store Queue

import { writable } from 'svelte/store';
import type { DialogSettings } from './types';

function dialogService() {
	const { subscribe, set, update } = writable<DialogSettings[]>([]);
	return {
		subscribe,
		/** Trigger - append to end of queue */
		trigger: (dialog: DialogSettings) =>
			update((dStore: DialogSettings[]) => {
				dStore.push(dialog);
				return dStore;
			}),
		/** Close - remove first item in queue */
		close: () =>
			update((dStore) => {
				if (dStore.length > 0) {
					dStore.shift();
				}
				return dStore;
			}),
		/** Clear - remove all items from queue */
		clear: () => set([])
	};
}

export const dialogStore = dialogService();
```

```
// Dialog Store Queue

import { writable } from 'svelte/store';
import type { DialogSettings } from './types';

function dialogService() {
	const { subscribe, set, update } = writable<DialogSettings[]>([]);
	return {
		subscribe,
		/** Trigger - append to end of queue */
		trigger: (dialog: DialogSettings) =>
			update((dStore: DialogSettings[]) => {
				dStore.push(dialog);
				return dStore;
			}),
		/** Close - remove first item in queue */
		close: () =>
			update((dStore) => {
				if (dStore.length > 0) {
					dStore.shift();
				}
				return dStore;
			}),
		/** Clear - remove all items from queue */
		clear: () => set([])
	};
}

export const dialogStore = dialogService();
```